tests/cpp/test_add_second_input_pass.cpp

```diff
@@ -0,0 +1,396 @@
+    auto eos_token = std::make_shared<v0::Constant>(element::i32, Shape{1}, std::vector<int32_t>{eos_token_id});
+
+    // IDs tensor (required as the last input for CombineSegment)
+    auto ids = std::make_shared<v0::Constant>(element::i32, Shape{2}, std::vector<int32_t>{0, 1});
+
+    // Prepare CombineSegment inputs: [begins1, ends1, data1, begins2, ends2, data2, ..., ids]
+    ov::OutputVector combine_inputs = {
+        // Main BPETokenizer outputs which went through Truncate
+        truncate[0], truncate[1], truncate[2],
+        // EOS token tensors
+        eos_begins, eos_ends, eos_token,
+        // IDs tensor
+        ids
+    };
+
+    auto CombineSegments = create_func("CombineSegments", {combine_inputs}, {});
+    model = std::make_shared<ov::Model>(OutputVector{CombineSegments}, ParameterVector{parameter_1});
+    
+    // Add a valid post_processor to model's rt_info
+    static const std::string PROCESSED_POST_PROCESSOR_NAME = "processed_post_processor_template";
+    // Signature for single input [first_seq, eos]
+    std::vector<int> input_signature = {-1, eos_token_id};
+    // Signature for pair [first_seq, eos_seq, second_seq, eos]
+    std::vector<int> pair_signature = {-1, eos_token_id, -1, eos_token_id};
+    
+    nlohmann::json post_processor = {
+        {"single", {{"ids", input_signature}}},
+        {"pair", {{"ids", pair_signature}, {"type_ids", std::vector<int>{0, 0, 1, 1}}}}
+    };
+
+    model->get_rt_info()[PROCESSED_POST_PROCESSOR_NAME] = post_processor.dump();
+
+    std::ostringstream pass_errors;
+    
+    ov::pass::Manager manager;
+    manager.register_pass<ov::pass::VisualizeTree>("graph_before.svg");
+    manager.register_pass<ov::genai::AddSecondInputPass>(shared_object_ov_tokenizers, pass_errors);
+    manager.register_pass<ov::pass::VisualizeTree>("graph_after.svg");
+    manager.run_passes(model);
+
+    // Check that resulting graph contains required fields.
+    ASSERT_EQ(model->get_parameters().size(), 2);
+    // Find combine segment op and assert that it has 2 extra inputs
+    auto results = model->get_results();
+    // ASSERT_EQ(results.size(), 1);
+    auto combine_segments_node = results[0]->get_input_node_shared_ptr(0);
+    ASSERT_NE(combine_segments_node, nullptr);
+    ASSERT_EQ(combine_segments_node->get_input_size(), combine_inputs.size() + 6); // 6 extra inputs for the second input and eos token
+}
+
+// Additional negative tests for AddSecondInputPass error cases
+
+// Helper to run the pass and return result (false if pass failed)
+static bool run_add_second_input_pass(std::shared_ptr<ov::Model> model, std::ostringstream& pass_errors) {
+    ov::pass::Manager manager;
+    manager.register_pass<ov::genai::AddSecondInputPass>(shared_object_ov_tokenizers, pass_errors);
+    try {
+        manager.run_passes(model);
+    } catch (...) {
+        // Some errors may throw, but pass should return false
+        return false;
+    }
+    return pass_errors.str().empty();
```

Comment:
It seems in case of normal execution `run_passes` will return `bool` so it's possible to `return manager.run_passes(model);`.
What is the reason to catch the error here? Wouldn't it hide unexpected fails?

Reply:
You are right, try/catch is redundant and masks potential problems. I removed it.

Regarding run_passes return flag it equivalent of checking if pass_errors is empty, because in any case if i exit from the pass leaving the model unchanged i a populate pass_errors with a relevant message, so i left this part unchanged.